Chap_API_Struct.tex

```diff
@@ -2622,6 +2622,54 @@ \subsubsection{Free a \refstruct{pmix_data_array_t} structure}
 \end{arglist}
 
 
+%%%%%%%%%%
+\subsection{Network Coordinate Structure}
+\declarestruct{pmix_coord_t}
+
+The \refstruct{pmix_coord_t} structure describes the network coordinates of a specified process.
+
+\versionMarker{4.0}
+\cspecificstart
+\begin{codepar}
+typedef struct pmix_coord \{
+    int x;
+    int y;
+    int z;
+\} pmix_coord_t;
+\end{codepar}
+\cspecificend
+
+All coordinate values shall be expressed as integers due to their units being defined in network devices and not physical distances. The coordinate is therefore an indicator of connectivity and not relative communication distance.
+
+\adviceimplstart
+Note that the \refstruct{pmix_coord_t} structure only defines the user-facing representation - it specifically does not imply nor mandate any requirement on how the coordinate data is to be stored within the \ac{PMIx} library. Implementers are free to store the coordinate in whatever format they choose.
+
+Note that a network coordinate is actually associated with a given network device - e.g., a particular \ac{NIC} on a node. Thus, while the network coordinate of a device must be unique, the coordinate may be shared by multiple processes on a node. If the node contains multiple network devices, then either the device closest to the binding location of a process should be used as its coordinate, or (if the process is unbound or its binding is not known) all devices on the node should be reported as a \refstruct{pmix_data_array_t} of \refstruct{pmix_coord_t} structures.
+\adviceimplend
```

Comment:
Maybe an additional note about multiple network planes.

> If a process is connected with multiple network planes it may have associated with it a coordinate on each of these planes. In which case the set of network coordinates should be reported as a \refstruct{pmix_data_array_t} of \refstruct{pmix_coord_t} structures. The caller may request a coordinate from a specific network plane by using the \refattr{PMIX_ALLOC_NETWORK_PLANE} attribute on the query.

Reply:
Good suggestion! Will add

Reply:
@jjhursey Please see what you think of the addition - did some minor word-smithing.

Reply:
I like the new wording. Thanks 👍